lib/internal/repl.js

```diff
@@ -127,7 +127,8 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
     }
 
     if (data) {
-      repl.history = data.split(/[\n\r]+/, repl.historySize);
+      repl.history = data.trim().split(/[\n\r]+/)
+        .reverse().slice(0, repl.historySize);
```

Comment:
> This PR changes it so that so that the number of lines stored in memory from the history file at start up is still limited, but the history file size itself is unbounded.

Is there any particular advantage to limiting the number of lines stored in memory if they are all going to be loaded (and saved?) anyway?